server/api/v2/types.go

```diff
@@ -22,5 +22,5 @@ type RunResponse struct {
 	// Stderr is standard error stream from running program
 	Stderr string `json:"stderr"`
 	// Error result from either compilation or runtime errors. Will be nil if both compilation and runtime were successful.
-	Error error `json:"error"`
+	Error string `json:"error"`
```

Comment:
One big thing I noticed is that the internal `json.Marshal` wasn't properly marshalling the error type. If I did something like 
```
fmt.Println("%v", err)
```
I would get the right error message, but once it was marshaled, it was returned as an object like 
```
{
...
error: {Stderr: null}
}
```
My solution was simply to call the `.Error()` method on the propagated errors, to turn them into strings before they're marshaled